### PR TITLE
Update docker, update test suite

### PR DESCRIPTION
**Pull request by sensei100**
What this PR does:
- updates docker files & makefile for running XSLT via Saxon locally & better XSpec commands
- migrates CI from Travis to CircleCI (part of department migration)
- updates existing tests to pass after couple iterations of pushing failed builds to master
- adds more docs in `README.md`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,39 +14,129 @@ The XSLT (all version 2) is written in 3 levels:
...
+All of the above generate test reports in the relevant test directory, e.g. `tests/schematron/xspec` or `tests/schematron/xspec`. These are HTML and XML versions of the output, which can be helpful to review if debugging a failing test (just open the directory in your web browser via a local filepath). These are always ignored by git, so never checked into GitHub.
 
-## Deployment to Combine
+## CI/CD
 
-CI/CD for this repository is handled by Travis.
+[CircleCI](https://circleci.com/gh/tulibraries/aggregator_mdx) manages running CI/CD for this application. It does the following, using the same Docker images we use locally:
+- build the Docker image with the repository files from that commit copied over;
+- run a version of `make test` with a command to fail if `Error running the test suite` or `Error compiling the test suite` appears in the test output.
+- [WIP] runs a coverage check that for every `.sch` file in `validations/` and `.xsl` in `transforms/` has a test file `[same filename].xspec` in `tests/`. CI will fail if this coverage is less than 80%.
```
Comment (sensei100) [inline README.md]: I really like this idea!

File: transforms/dplah.xsl [hunk 2]
```diff
@@ -129,15 +137,15 @@
         </xsl:if>
     </xsl:template>
 
-    <!-- URL 
+    <!-- URL
```
Comment (sensei100) [inline transforms/dplah.xsl]: Does this need the closing --> ?

File: fixtures/schematron/invalid_empty_dataprovider1.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<oai_dc:dc xmlns:dc="http://purl.org/dc/elements/1.1/"
+  xmlns:dcterms="http://purl.org/dc/terms/"
+  xmlns:edm="http://www.europeana.eu/schemas/edm/"
+  xmlns:oai_dc="http://www.openarchives.org/OAI/2.0/oai_dc/"
+  xmlns:dpla="http://dp.la/about/map/" xmlns:schema="http://schema.org"
+  xmlns:oai="http://www.openarchives.org/OAI/2.0/"
+  xmlns:oai_qdc="http://worldcat.org/xmlschemas/qdc-1.0/">
+  <dcterms:title/>
```
Comment (lfinnigan) [inline fixtures/schematron/invalid_empty_dataprovider1.xml]: working on a PR to update xspec to fix these empty titles

Comment (lfinnigan) [reply fixtures/schematron/invalid_empty_dataprovider1.xml]: putting the brakes on my PR until these get merged

**Pull request merged**